core/app/modules/onboarding/assets/js/pages/account.js

```diff
@@ -160,7 +168,19 @@ export default function Account() {
 		if ( 'initial' !== dataSharingAjaxState.status ) {
 			if ( 'success' === dataSharingAjaxState.status && dataSharingAjaxState.response?.usageDataShared ) {
 				elementorCommon.config[ 'event-tracker' ].isUserDataShared = true;
-				updateState( { isUsageDataShared: true } );
+
+				// If connect was successful
+				if ( state.isLibraryConnected ) {
+					updateState( { isUsageDataShared: true } );
```

Comment:
He may be connected without sharing data

Reply:
Yes. This line is only called if the "usage data sharing" checkbox was marked when the user clicked "Create Account" or "Skip", and the Ajax request for updating the DB option was successful.